parser-typechecker/src/Unison/PrintError.hs

```diff
@@ -1115,10 +1115,26 @@ prettyParseError ::
   String ->
   Parser.Err v ->
   Pretty ColorText
-prettyParseError s = \case
-  P.TrivialError _ (LexerError ts e) _ -> go e
+prettyParseError s e =
+  mconcat (fst <$> renderParseErrors s e) <> lexerOutput
+  where
+    lexerOutput :: Pretty (AnnotatedText a)
+    lexerOutput =
+      if showLexerOutput
+        then "\nLexer output:\n" <> fromString (L.debugLex' s)
+        else mempty
+
+renderParseErrors ::
```

Comment:
New combinator that renders a parser error into a list of errors and the ranges within the file to which each of those errors applies.

These ranges will be used to tell the LSP where to show diagnostic errors.